File: Novels/packages/6-mobile-android/StoryReader/app/src/main/java/com/storysphere/storyreader/model/Subscription.kt
package com.storysphere.storyreader.model

data class Subscription(
    val id: String,
    val userId: String,
    val planId: String,
    val planName: String,
    val planType: SubscriptionPlanType,
    val status: SubscriptionStatus,
    val startDate: Long,
    val endDate: Long? = null, // null for unlimited
    val autoRenew: Boolean = true,
    val createdAt: Long = System.currentTimeMillis(),
    val updatedAt: Long = System.currentTimeMillis()
)

enum class SubscriptionPlanType {
    ALL_YOU_CAN_READ,  // Netflix-style unlimited reading
    VIP_LOYALTY        // VIP levels with discounts
}

enum class SubscriptionStatus {
    ACTIVE,
    CANCELLED,
    EXPIRED,
    PENDING
}

data class SubscriptionPlan(
    val id: String,
    val name: String,
    val type: SubscriptionPlanType,
    val price: Int, // Monthly price in points
    val features: List<String>,
    val duration: Int? = null // Duration in days, null for unlimited
)

File: Novels/packages/5-mobile-ios/StoryReader/Model/Subscription.swift
import Foundation

// Subscription model - Membership/Subscription plans
struct Subscription: Codable, Identifiable {
    let id: String
    let userId: String
    let planId: String
    let plan: MembershipPlan?
    let status: SubscriptionStatus
    let startDate: Date
    let endDate: Date?
    let autoRenew: Bool
    let createdAt: Date
    let updatedAt: Date
    
    enum SubscriptionStatus: String, Codable {
        case active
        case cancelled
        case expired
    }
    
    enum CodingKeys: String, CodingKey {
        case id
        case userId = "user_id"
        case planId = "plan_id"
        case plan
        case status
        case startDate = "start_date"
        case endDate = "end_date"
        case autoRenew = "auto_renew"
        case createdAt = "created_at"
        case updatedAt = "updated_at"
    }
}

// MembershipPlan model
struct MembershipPlan: Codable, Identifiable {
    let id: String
    let name: String
    let price: Double // In real currency (CNY)
    let coinsGranted: Int // Immediate coin grant
    let dailyBonus: Int // Daily login bonus
    let billingPeriod: BillingPeriod
    let isActive: Bool
    let createdAt: Date
    let updatedAt: Date
    
    enum BillingPeriod: String, Codable {
        case monthly
        case yearly
    }
    
    enum CodingKeys: String, CodingKey {
        case id
        case name
        case price
        case coinsGranted = "coins_granted"
        case dailyBonus = "daily_bonus"
        case billingPeriod = "billing_period"
        case isActive = "is_active"
        case createdAt = "created_at"
        case updatedAt = "updated_at"
    }
}

